programs/bpf_loader/src/syscalls/cpi.rs

```diff
@@ -608,6 +612,10 @@ where
         ))?;
     accounts.push((*program_account_index, None));
 
+    // unwrapping here is fine: we're in a syscall and the method below fails
+    // only outside syscalls
+    let orig_data_lens = invoke_context.get_orig_account_lengths().unwrap();
```

Comment:
I agree that it is safe, but why not `map_err(EbpfError::SyscallError)` for consistency?

Reply:
Because it's good to be able to assert things, so it's easier to reason about program flow. If you see an unwrap, you know immediately that the error path is not expected. If you have a bunch of map_err scattered everywhere just in case (like we do in some places), it's harder to refactor code in the future once you have forgotten about all the possible conditions.

Ideally, I think we should have defensive code only at the boundaries where we interact with the external world. In implementation code, we should assert/unwrap so we catch logic errors early. This does require a high test coverage to make sure that we do catch bad inputs at the boundaries. The CPI code does have a high coverage now.